### PR TITLE
Handle EOFError in FileSystemCache

### PR DESCRIPTION
**Pull request by northernSage**
Addresses #21

- [x] handle `EOFError`
- [x] add change log
- [x] add warning about corrupted cache files
- [x] add tests covering handling of empty cache files
- [x] add extra check to `FileSystemCache.set` to ensure empty files are not being created

### CODE REVIEW

File: src/cachelib/__init__.py [hunk 1]
```diff
@@ -16,3 +16,9 @@
     "UWSGICache",
 ]
 __version__ = "0.2.0"
+
+
+import logging
+from logging import NullHandler
+
+logging.getLogger(__name__).addHandler(NullHandler())
```
Comment (davidism) [inline src/cachelib/__init__.py:24]: Shouldn't need to add a null handler. My understanding is that libraries should do no logging configuration, only emit messages.

Comment (northernSage) [reply src/cachelib/__init__.py:24]: @davidism The idea was preventing events from being outputted to `sys.stderr` in the absence of logging configuration as pointed out in the [docs](https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library). I'm OK with it If you think the default behaviour is desirable. Should we just remove this then?
